docs: rename catapult-server to catapult-client in docs (#189)
symbol/catapult-client: README.md

# catapult-client

> [![docs](badges/docs--green.svg)](https://nemtech.github.io)

Reviewer: ```suggestion
[![docs](badges/docs--green.svg)](https://symbol.github.io)
```

Symbol-based networks rely on nodes to provide a trustless, high-performance, and secure blockchain platform.

These nodes are deployed using [catapult-client] software, a C++ rewrite of the previous Java-written [NEM] distributed ledger that has been running since 2015.

> Learn more about the protocol by reading the [whitepaper] and the  [developer documentation].

Reviewer: ```suggestion
Learn more about the protocol by reading the [technical reference] and the  [developer documentation].
```

## Package Organization

catapult-client code is organized as follows:

| Folder name | Description |
| -------------|--------------|
| /extensions | Modules that add features to the bare catapult-client. These capabilities are all optional because none of them impact consensus. |

> | /external | External dependencies that are built with the server. |

Reviewer: ```suggestion
| /external | External dependencies that are built with the client. |
```

[developer documentation]: https://nemtech.github.io
[Forum]: https://forum.nem.io/c/announcement
[issues]: https://github.com/symbol/catapult-client/issues
[slack]: https://join.slack.com/t/nem2/shared_invite/enQtMzY4MDc2NTg0ODgyLWZmZWRiMjViYTVhZjEzOTA0MzUyMTA1NTA5OWQ0MWUzNTA4NjM5OTJhOGViOTBhNjkxYWVhMWRiZDRkOTE0YmU
[catapult-client]: https://github.com/symbol/catapult-client
[symbol-rest]: https://github.com/nemtech/catapult-rest
[Service Bootstrap]: https://github.com/symbol/symbol-bootstrap
[nem]: https://nem.io

> [whitepaper]: https://nemtech.github.io/catapult-whitepaper/main.pdf

Reviewer: ```suggestion
[technical reference]: https://symbol.github.io/symbol-technicalref/main.pdf
```
